src/WorkerHandler.js

```diff
@@ -399,17 +418,46 @@ WorkerHandler.prototype.exec = function(method, params, resolver, options) {
   var me = this;
   return resolver.promise.catch(function (error) {
     if (error instanceof Promise.CancellationError || error instanceof Promise.TimeoutError) {
+      me.tracking[id] = {
+        id,
+        resolver: Promise.defer()
+      };
+      
       // remove this task from the queue. It is already rejected (hence this
       // catch event), and else it will be rejected again when terminating
       delete me.processing[id];
 
-      // terminate worker
-      return me.terminateAndNotify(true)
-        .then(function() {
-          throw error;
-        }, function(err) {
-          throw err;
-        });
+      me.tracking[id].resolver.promise = me.tracking[id].resolver.promise.catch(function(err) {
+        return me.terminateAndNotify(true)
+          .then(function() { 
+            throw err;
+          }, function(err) {
+            throw err;
+          }); 
+      });
+      
+      me.worker.send({
+        id,
+        method: CLEANUP_METHOD_ID 
+      });
+
+      /**
+       * Sets a timeout to reject the cleanup operation if the message sent to the worker
```

Comment:
Thanks for this explanation.

Can you also write documentation about this new feature in the README.md?